mkosi/__init__.py

```diff
@@ -3675,11 +3699,12 @@ def insert_partition(
         old_table = PartitionTable.read(loopdev)
     else:
         # No partition table yet? Then let's fake one...
-        old_table = PartitionTable.empty()
+        old_table = PartitionTable.empty(args.gpt_first_lba)
 
-    blob_size = roundup512(os.stat(blob.name).st_size)
+    blob_size = roundup(os.stat(blob.name).st_size, 512)
```

Comment:
we should probably start rounding up partition sizes, and start offsets to multiples of 4K too. it's where things are really going with today's HDDs and it makes a ton of sense to align disk sector size and memory page sizes. and the 3.5K this might waste here and there are probably nothing to be concerned about, regardless if we are talking about tiny initrd extensions or anything else.

Reply:
Yep. I also changed partition alignment to 4k, so rounding to 512 will not make any difference.

Reply:
you did? don't see it?

Reply:
I changed the partition alignment, not the size of the blob…